ir/instr.cpp

```diff
@@ -1247,6 +1249,12 @@ StateValue FnCall::toSMT(State &s) const {
   auto ret = s.addFnCall(fnName_mangled.str(), move(inputs), move(ptr_inputs),
                          out_types, !(flags & NoRead), !(flags & NoWrite),
                          flags & ArgMemOnly, move(returned_val));
+
+  if (flags & NoReturn) {
+    // TODO: Even if a function call doesn't have noreturn, it can possibly
+    // exit. Relevant bug: https://bugs.llvm.org/show_bug.cgi?id=27953
+    s.addNoReturn(expr(false));
```

Comment:
I'll expand this after #323 is reviewed; to do this, addFnCall should be fixed

Reply:
For fixing addFnCall, I'll make a separate PR after this is merged

Reply:
that's why you leave a condition there? you want to pass a non-deterministic boolean in the remaining cases?

Reply:
We should maintain a boolean var saying whether the function call exited or not, and mkAxiom() should encode a relation saying that if inputs are the same their exited-ness are equivalent.